onnxruntime/core/graph/graph_proto_serializer.cc

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/graph/graph_proto_serializer.h"
+
+namespace onnxruntime {
+
+void GraphViewerToProto(const GraphViewer& graph_view, ONNX_NAMESPACE::GraphProto& graph_proto, bool include_initializer) {
+  graph_proto.set_name(graph_view.Name());
+  graph_proto.set_doc_string(graph_view.Description());
+
+  for (const auto* input_arg : graph_view.GetInputsIncludingInitializers()) {
+    *(graph_proto.mutable_input()->Add()) = input_arg->ToProto();
+  }
+
+  for (const auto* output_arg : graph_view.GetOutputs()) {
+    *(graph_proto.mutable_output()->Add()) = output_arg->ToProto();
+  }
+
+  for (const auto* value_info : graph_view.GetValueInfo()) {
+    *(graph_proto.mutable_value_info()->Add()) = value_info->ToProto();
+  }
+
+  // add the NodeArg info for outer scope NodeArgs so we capture the type information
+  for (const auto& name : graph_view.GetOuterScopeNodeArgNames()) {
+    auto* node_arg = graph_view.GetNodeArg(name);
+    ORT_ENFORCE(node_arg, "Outer scope node arg name '" + name + "'was added but does not exist. ");
+    *(graph_proto.mutable_value_info()->Add()) = node_arg->ToProto();
+  }
+
```

Comment:
Do you need a flag here for when it's being converted to a stand-alone graph, that when true results in all the outer scope node args being made explicit graph inputs? That may be required for TRT.

Reply:
add the out of scope argument flag.

Reply:
Still not clear to me if TRT would potentially require the outer scope args to also be explicit graph inputs, but simple to add that later if needed. 

AFAIK no other EP would require that functionality.